### DISCUSSION TITLE
Adding shapely as an optional dependancy

### DISCUSSION THREAD

**Discussion by briochh** [pypest/pyemu#392] [General]
@rhugman-intera has some methods for locating pilot points on grid, leaning on shapely:  https://github.com/pypest/pyemu/pull/387#issuecomment-1383803709 . I don't see any obvious reason we can have shapely as an optional dependancy (we might already be some other methods using it?). Thoughts @jtwhite79?

**Comment by jtwhite79** [comment]
Can we isolate that dependency so that it's only imported if someone calls that function?

**Reply by rhugman** [reply]
Sure - imports can be called at the function level?

**Reply by jtwhite79** [reply]
I think so - so we just try-catch on the `import shapely` within the fxn that needs it.  That would let those who dont need that dep continue without it...I think FloPy used to do this also (or maybe still does in a few places?)
